### PR TITLE
upgrade eslint to version 8, add stricter rules, and add templates for next and angular

### PR DESCRIPTION
**Pull request by GAlexIHU**
Possible TODO:
- next and angular templates are not super details
- still have to try out locally on different projects to see how it feels

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-[![npm version](http://img.shields.io/npm/v/@codingsans/eslint-config.svg?style=flat)](https://npmjs.org/package/@codingsans/eslint-config "View this project on npm")
+[![npm version](http://img.shields.io/npm/v/@codingsans/eslint-config.svg?style=flat)](https://npmjs.org/package/@codingsans/eslint-config 'View this project on npm')
 [![MIT license](http://img.shields.io/badge/license-MIT-brightgreen.svg)](http://opensource.org/licenses/MIT)
 
-
 # eslint-config-codingsans
+
 Sharable eslint config created and used by CodingSans
 
 ## ESLINT Version
```
Comment (latotty) [inline README.md:8]: This config created for eslint 8

File: angular-recommended.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  extends: ['./ts-recommended.js', ''],
```
Comment (latotty) [inline angular-recommended.js]: `extends: ['./ts-recommended.js'],`

File: package.json [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
+  },
   "keywords": [
     "eslint",
     "eslintconfig"
   ],
-  "scripts": {
-    "lint": "eslint '*.js'",
-    "check-generated-config": "eslint -c index.js --print-config typescript-recommended.js > generated.typescript-recommended.json"
+  "engines": {
+    "node": "^14.18.0 || >=16.0.0"
```
Comment (latotty) [inline package.json:15]: node 15?

File: tsconfig.base.json [hunk 4]
```diff
@@ -2,40 +2,40 @@
   "compilerOptions": {
     /* Basic Options */
     // https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping
-    "lib": ["ES2019"],
+    "lib": ["ESNext"],
     "module": "commonjs",
-    "target": "ES2019",
+    "target": "ESNext",
```
Comment (latotty) [inline tsconfig.base.json]: that is too wide, fix it to node14 (ES2020)

**Review by latotty — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (GAlexIHU) [reply package.json:15]: https://eslint.org/docs/latest/user-guide/migrating-to-8.0.0#drop-old-node

**Pull request merged**